### PR TITLE
[SYCL] Unroll several loops in __init method accessor class

### PR DESCRIPTION
**Pull request by MrSidims**
It might improve performance on several platforms.

Co-author: Mark Mendell <mark.p.mendell@intel.com>
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
This clang-format suggestion is weird.

**Review by s-kanaev — APPROVED** [review]

File: sycl/include/CL/sycl/accessor.hpp [hunk 1]
```diff
@@ -851,7 +851,7 @@ class accessor :
   void __init(ConcreteASPtrType Ptr, range<AdjustedDim> AccessRange,
               range<AdjustedDim> MemRange, id<AdjustedDim> Offset) {
     MData = Ptr;
-    for (int I = 0; I < AdjustedDim; ++I) {
+    [[clang::loop_unroll]] for (int I = 0; I < AdjustedDim; ++I) {
```
Comment (bader) [inline sycl/include/CL/sycl/accessor.hpp:854]: Why did you decide to use clang attribute?
`#pragma` seems to be more portable.

Comment (MrSidims) [reply sycl/include/CL/sycl/accessor.hpp:854]: To make a claim, that we have done something useful, when implemented it :)

Comment (MrSidims) [reply sycl/include/CL/sycl/accessor.hpp:854]: I don't think, that `pragma` spelling is more portable: for example GCC has `pragma GCC unroll`, not `pragma unroll`.

Comment (bader) [reply sycl/include/CL/sycl/accessor.hpp:854]: Isn't test added with implementation enough "to make a claim"?

**Comment by MrSidims** [comment]
@s-kanaev ping

**Review by s-kanaev — APPROVED** [review]

**Pull request merged**